src/main/java/edu/carroll/dogtag/web/controller/RegisterController.java

```diff
@@ -0,0 +1,113 @@
+     * @param result       this is to check errors on the templates and display the error message
+     * @param attrs        used to redirect user info to the register success pages to display the registered
+     *                     name entered.
+     * @return there is any errors within the form or the template itself
+     * if the user is found, email is found, or required entry are not met
+     * and error is returned for using a different username or email.
+     * If any errors of occur the register is page is reloaded. If no errors then
+     * the control check if a user or email exist in the database.
+     * Once the checks are complete it allows the controller to submit the information to the
+     * RegisterService and then is sent to the register success page.
+     */
+    @PostMapping("/register")
+    public String registerPost(@Valid @ModelAttribute RegisterForm registerForm, BindingResult result, Model attrs) {
+        log.info("Logging info for {} registration", registerForm.getUser());
+        if (result.hasErrors()) {
+            log.error("{} could not be registered and was not stopped by @Notnull and @Size", registerForm.getUser());
+            return "register";
+        }
+
+        if (registerService.userExists(registerForm.getUser()) && registerService.emailExists(registerForm.getEmail())) {
+            result.addError(new ObjectError("email", "Can not register: Please use a different Email"));
+            log.info("Email {} exists in database already", registerForm.getEmail());
+            result.addError(new ObjectError("user", "Can not register: Please use a different User Name"));
+            log.info("User {} exists in database already", registerForm.getUser());
+            return "register";
+        }
+
+        if (registerService.userExists(registerForm.getUser())) {
+            result.addError(new ObjectError("user", "Can not register: Please use a different User Name"));
+            log.info("User {} exists in database already", registerForm.getUser());//show what user failed to register
+            return "register";
+        }
+
+        if (registerService.emailExists(registerForm.getEmail())) {
+            result.addError(new ObjectError("email", "Can not register: Please use a different Email"));
+            log.info("Email {} exists in database already", registerForm.getEmail());//show what email failed vaidation
+            return "register";
+        }
+
+        Login userRegister = new Login();
+        userRegister.setUser(registerForm.getUser());
+        userRegister.setEmail(registerForm.getEmail());
+        userRegister.setPassword(registerForm.getPassword());
+        if (!registerService.register(userRegister)) {
+            return "redirect:/register";
+        }
+        registerService.register(userRegister);
+        attrs.addAttribute("user", registerForm.getUser());
```

Comment:
You're setting an attribute (user), but the registerSuccess doesn't use the parameter.  The user of the attribute should be removed since it's unused.